staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -132,15 +132,19 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (r
 		return nil, fmt.Errorf("failed to build manager identifier: %v", err)
 	}
 
+	originalManagedFields := managed.Fields[manager]
+
 	// TODO(apelisse) use the first return value when unions are implemented
 	_, managed.Fields, err = f.updater.Update(liveObjTyped, newObjTyped, apiVersion, managed.Fields, manager)
 	if err != nil {
 		return nil, fmt.Errorf("failed to update ManagedFields: %v", err)
 	}
 	managed.Fields = f.stripFields(managed.Fields, manager)
 
-	// Update the time in the managedFieldsEntry for this operation
-	managed.Times[manager] = &metav1.Time{Time: time.Now().UTC()}
+	// Update the time in the managedFieldsEntry for this operation if the fieldset changed
+	if originalManagedFields == nil || !managed.Fields[manager].Set().Difference(originalManagedFields.Set()).Empty() {
```

Comment:
This would mean that we only update the time on these entries when the field set changes, not when the values change of something which is already owned.

Also, we don't need to check the reverse difference because an updater's fieldset can only grow by doing an update operation.

Reply:
> an updater's fieldset can only grow by doing an update operation.

Can you elaborate on that? What if you remove an optional field?

Reply:
Removing a field or list item is something I didn't think of. We can check both directions of the difference then. But I wasn't sure if this is even the direction we want to go for when to update the time

Reply:
Thinking about it more, we could probably do this instead:
`if c, err := liveObjTyped.Compare(newObjTyped); err == nil && !c.IsSame() {`